uportal-war/src/main/java/org/jasig/portal/groups/pags/dao/jpa/JpaPersonAttributesGroupDefinitionDao.java

```diff
@@ -86,64 +117,37 @@ public void deletePersonAttributesGroupDefinition(IPersonAttributesGroupDefiniti
-        CriteriaQuery<PersonAttributesGroupDefinitionImpl> criteriaQuery = 
-                criteriaBuilder.createQuery(PersonAttributesGroupDefinitionImpl.class);
-        Root<PersonAttributesGroupDefinitionImpl> root = criteriaQuery.from(PersonAttributesGroupDefinitionImpl.class);
-        ParameterExpression<String> nameParameter = criteriaBuilder.parameter(String.class);
-        criteriaQuery.select(root).where(criteriaBuilder.equal(root.get("name"), nameParameter));
-        TypedQuery<PersonAttributesGroupDefinitionImpl> query = this.getEntityManager().createQuery(criteriaQuery);
+        TypedQuery<PersonAttributesGroupDefinitionImpl> query = this.createCachedQuery(groupDefinitionByNameQuery);
         query.setParameter(nameParameter, name);
-        Set<IPersonAttributesGroupDefinition> groups = new HashSet<IPersonAttributesGroupDefinition>();
-        for (IPersonAttributesGroupDefinition group: query.getResultList()) {
-            groups.add(group);
-        }
+        Set<IPersonAttributesGroupDefinition> groups = new HashSet<IPersonAttributesGroupDefinition>(query.getResultList());
         if(groups.size() > 1) {
             logger.error("More than one PAGS Group found for name: {}", name);
         }
         return groups;
     }
 
-    @PortalTransactional
     @Override
     public Set<IPersonAttributesGroupDefinition> getParentPersonAttributesGroupDefinitions(IPersonAttributesGroupDefinition group) {
-        CriteriaBuilder criteriaBuilder = this.getEntityManager().getCriteriaBuilder();
-        CriteriaQuery<PersonAttributesGroupDefinitionImpl> criteriaQuery =
-                criteriaBuilder.createQuery(PersonAttributesGroupDefinitionImpl.class);
-        Root<PersonAttributesGroupDefinitionImpl> root = criteriaQuery.from(PersonAttributesGroupDefinitionImpl.class);
-
-        Join<PersonAttributesGroupDefinitionImpl, PersonAttributesGroupDefinitionImpl> members =
-                root.join(PersonAttributesGroupDefinitionImpl_.members);
-        ParameterExpression<String> nameParameter = criteriaBuilder.parameter(String.class);
-        criteriaQuery.where(criteriaBuilder.equal(members.get(PersonAttributesGroupDefinitionImpl_.name), nameParameter));
-        TypedQuery<PersonAttributesGroupDefinitionImpl> query = this.getEntityManager().createQuery(criteriaQuery);
+        TypedQuery<PersonAttributesGroupDefinitionImpl> query = this.createCachedQuery(parentGroupDefinitionsQuery);
         query.setParameter(nameParameter, group.getName());
-
-        Set<IPersonAttributesGroupDefinition> parents = new HashSet<IPersonAttributesGroupDefinition>();
-        for (IPersonAttributesGroupDefinition parent : query.getResultList()) {
-            parents.add(parent);
-        }
-        return parents;
+        Set<IPersonAttributesGroupDefinition> rslt = new HashSet<IPersonAttributesGroupDefinition>(query.getResultList());
+        return rslt;
     }
 
-    @PortalTransactional
```

Comment:
Do we need @PortalTransactionalReadOnly here?  (I'm pretty sure we don't need @PortalTransactional).